cpp/src/join/conditional_join_kernels.cuh

```diff
@@ -67,20 +67,21 @@ __global__ void compute_conditional_join_output_size(
     &intermediate_storage[threadIdx.x * device_expression_data.num_intermediates];
 
   std::size_t thread_counter{0};
-  cudf::size_type const start_idx      = threadIdx.x + blockIdx.x * block_size;
-  cudf::size_type const stride         = block_size * gridDim.x;
-  cudf::size_type const left_num_rows  = left_table.num_rows();
-  cudf::size_type const right_num_rows = right_table.num_rows();
-  auto const outer_num_rows            = (swap_tables ? right_num_rows : left_num_rows);
-  auto const inner_num_rows            = (swap_tables ? left_num_rows : right_num_rows);
+  cudf::thread_index_type const start_idx      = threadIdx.x + blockIdx.x * block_size;
+  cudf::thread_index_type const stride         = thread_index_type{block_size} * gridDim.x;
+  cudf::thread_index_type const left_num_rows  = left_table.num_rows();
+  cudf::thread_index_type const right_num_rows = right_table.num_rows();
```

Comment:
FYI: You don't _have_ to upcast these values, because normal integer casting rules for the comparisons (`outer_row_index < outer_num_rows`) will make sure that they're upcast.

Reply:
Figured it was safer in case there was any arithmetic I missed.